.github/workflows/static-analysis.yml

```diff
@@ -1,6 +1,6 @@
 name: Static Analysis
 
-on: pull_request
+on: [pull_request, push]
```

Comment:
I think we should trigger push only for the master branch, to avoid that we hit the concurrent jobs limit once more checks are migrated to GitHub actions.
https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits